### PR TITLE
Prop method PR updated

### PR DESCRIPTION
**Pull request by rwaldin**
Updated version of PR #516 rebased against latest master.  I think I've address feedback to the original PR, added a bit more to the implementation of .prop('style'), plus tests.  For issue #490.

### CODE REVIEW

File: lib/api/attributes.js [hunk 1]
```diff
@@ -73,6 +73,75 @@ exports.attr = function(name, value) {
...
+};
+
+exports.prop = function (name, value) {
+  var i = 0,
+      property;
+
+  if (typeof name === 'string' && value === undefined) {
+
+    switch (name) {
+      case 'style':
```
Comment (jugglinmike) [inline lib/api/attributes.js]: Cheerio's responsibilities stop at the jQuery API; mimicking DOM API's in this library is a slipperly slope. For instance, why `item()` but not `removeProperty()`? A more complete implementation might find it's home in [the DOMHandler library](https://github.com/fb55/domhandler) which would trickle down to Cheerio. It feels like an over-extension to insert a partial implementation here.

Comment (rwaldin) [inline lib/api/attributes.js]: Sounds reasonable.  Updated PR to remove item() method and tests.

File: lib/api/attributes.js [hunk 2]
```diff
@@ -73,6 +73,71 @@ exports.attr = function(name, value) {
...
+        property = this.css();
+
+        _.each(property, function (v, p) {
```
Comment (rwaldin) [inline lib/api/attributes.js]: If you prefer, I can remove the setting of property names and length on the property object...  Or to go even further, remove the special handling of getting 'style' and 'tagName|nodeName' altogether.  Any of those options would be reasonable in order to get a prop() method.

**Comment by ApXaHgheJI** [comment]
When the prop will work ?

**Comment by fb55** [comment]
@rwaldin Sorry for the delay – and thanks!

**Pull request merged**